Sources/MongoKitten/AggregateStage.swift

```diff
@@ -36,9 +39,26 @@ public struct AggregateBuilderStage {
     }
     
     public static func project(_ projection: Projection) -> AggregateBuilderStage {
+        func valueUnwrapper(_ value: Primitive) -> WireVersion?{
+            switch value {
+            case let value as Int32:
+                if value == 0 { return .mongo3_4 } // indicates excluded fields
+                return nil
+            case let value as String:
+                if value == "$$REMOVE" { return .mongo3_6 } // indicates conditionally excluded fields
+                return nil
+            case let value as Document:
+                return valueUnwrapper(value)
+            default:
+                return nil
+            }
+        }
+        
+        let minimalVersionRequired: WireVersion? = projection.document.values.compactMap{ valueUnwrapper($0) }.min()
+        
         return AggregateBuilderStage(document: [
             "$project": projection.document
-        ])
+        ], minimalVersionRequired)
```

Comment:
If, as we discussed in slack, `AggregateStage` is redefined as a protocol, we can make the minimum version a computed property rather than a stored property. Doing so will allow us to skip any heavy operations, such as checking what MongoDB version is required for a projection. The performance cost mainly matters in production, and the `#if DEBUG` statement would disable the logging on production. Therefore, we hit two birds with one stone, assuming this becomes a setup with computed properties.

I'd also like to move the `minimalVersionRequired` calculations into a computed property in the `Projection` type.